smartsim/_core/launcher/colocated.py

```diff
@@ -67,9 +67,11 @@ def write_colocated_launch_script(
         # STDOUT of the job
         if colocated_settings["debug"]:
             script_file.write("export SMARTSIM_LOG_LEVEL=debug\n")
-
-        script_file.write(f"{colocated_cmd}\n")
-        script_file.write("DBPID=$!\n\n")
+        script_file.write(f"db_stdout=$({colocated_cmd})\n")
+        # pylint: disable=anomalous-backslash-in-string
+        script_file.write(
+            "DBPID=$(echo $db_stdout | sed -n 's/.*__PID__\([0-9]*\)__PID__.*/\\1/p')\n"
+        )
```

Comment:
Small thing I saw during stand-up: Consider making this a r-string or explicitly escaping the anomalous backslashes rather than silencing the pylint error:
```suggestion
        script_file.write(
            "DBPID=$(echo $db_stdout | sed -n 's/.*__PID__\\([0-9]*\\)__PID__.*/\\1/p')\n"
        )
```